src/views/shared/workflows-header/workflows-header.tsx

```diff
@@ -0,0 +1,113 @@
+'use client';
+import { useState } from 'react';
+
+import { Segment, SegmentedControl } from 'baseui/segmented-control';
+
+import usePageFilters from '@/components/page-filters/hooks/use-page-filters';
+import PageFiltersFields from '@/components/page-filters/page-filters-fields/page-filters-fields';
+import PageFiltersSearch from '@/components/page-filters/page-filters-search/page-filters-search';
+import PageFiltersToggle from '@/components/page-filters/page-filters-toggle/page-filters-toggle';
+import {
+  type PageQueryParamSetterValues,
+  type PageQueryParamKeys,
+  type PageQueryParams,
+} from '@/hooks/use-page-query-params/use-page-query-params.types';
+
+import WORKFLOWS_SEARCH_DEBOUNCE_MS from './config/workflows-search-debounce-ms.config';
+import { overrides, styled } from './workflows-header.styles';
+import { type Props } from './workflows-header.types';
+import WorkflowsQueryInput from './workflows-query-input/workflows-query-input';
+import WorkflowsQueryLabel from './workflows-query-label/workflows-query-label';
+
+export default function WorkflowsHeader<
+  P extends PageQueryParams,
+  I extends PageQueryParamKeys<P>,
+  S extends PageQueryParamKeys<P>,
+  Q extends PageQueryParamKeys<P>,
+>({
+  pageQueryParamsConfig,
+  pageFiltersConfig,
+  inputTypeQueryParamKey,
+  searchQueryParamKey,
+  queryStringQueryParamKey,
+  refetchQuery,
+  isQueryRunning,
+}: Props<P, I, S, Q>) {
+  const [areFiltersShown, setAreFiltersShown] = useState(false);
+
+  const { resetAllFilters, activeFiltersCount, queryParams, setQueryParams } =
+    usePageFilters({
+      pageFiltersConfig,
+      pageQueryParamsConfig,
+    });
+
+  const inputType = queryParams[inputTypeQueryParamKey];
+  const query = queryParams[queryStringQueryParamKey];
+
+  return (
+    <styled.HeaderContainer>
+      <styled.InputContainer>
+        <SegmentedControl
+          activeKey={inputType}
+          onChange={({ activeKey }) => {
+            setQueryParams(
+              {
+                [inputTypeQueryParamKey]:
+                  activeKey === 'query' ? 'query' : 'search',
+              } as Partial<PageQueryParamSetterValues<P>>,
+              { replace: false, pageRerender: true }
```

Comment:
Why do we need to treat the input type different than other filters by having `pageRerender:true` ?
Also is this toggle state important enough to keep track of it in the history stack ?

Reply:
1) If we don't force a page rerender, the table still shows the old workflows from search mode.
2) Not particularly, but I don't see the difference either way. This was the old behaviour added in https://github.com/cadence-workflow/cadence-web/pull/738/files#diff-961085652f90fc92a5915251092215a3f7138044a3ba9d49d7f8c4930b413e38R34 so I just retained it.

Reply:
I tested moving between both tabs with `pageRerender: false` and seems to be working normally. If you confirm it is working this can save us requesting the RSC again from server.